scene/3d/skeleton_3d.cpp

```diff
@@ -853,6 +855,20 @@ Ref<SkinReference> Skeleton3D::register_skin(const Ref<Skin> &p_skin) {
 	return skin_ref;
 }
 
+// helper functions
+Transform Skeleton3D::bone_transform_to_world_transform(Transform p_bone_transform) {
+	Transform return_val = Transform();
+	return_val.origin = get_global_transform().xform(p_bone_transform.origin);
+	return_val.basis = get_global_transform().basis * p_bone_transform.basis;
```

Comment:
How is this different from get_global_transform() * p_bone_transform?

Also this appears to be the global transform of the skeleton and the local bone transform? 

This isn't the transform needed to go from the skeleton node to the bone from the root spatial node transform.

Reply:
Also, above information can be calculated when the process order of the bone struct is calculated. then the global transform of the skeleton can be fetched.

TL;DR, why isn't this in the bone struct?

Reply:
> How is this different from get_global_transform() * p_bone_transform?

Hmm, it might not be any different, thinking about it. If that snippet works, then that would be ideal. I will need to test! I am all for using simpler, easier to understand code :+1: 

> Also this appears to be the global transform of the skeleton and the local bone transform?

Here's what the function is intended to be used for, which hopefully can help better explain more about it:

The function is for taking a bone global pose (`global_pose`, `global_pose_override`, or one of the local poses converted to a global pose) and returns a transform that is the bone's position in the scene. `bone_transform_to_world_transform` is so you can do computations with the transform as if it is a global transform received from a Node3D. Then, when you are finished using the transform in the scene and want to apply it to the bone transform, you can use `world_transform_to_bone_transform` to get the transform as a global pose, when then you can apply.

Hopefully that helps clear things up a bit. It is kinda hard to explain, but it can be extremely useful to have when you need it.
I"m not totally happy with the naming though. I'd prefer to change the name(s) of the functions in #39353 rather than here though, just because said PR builds on top of this one. I am thinking a more descriptive name would be something like `global_pose_to_world_transform` and `world_transform_to_global_pose`.

> TL;DR, why isn't this in the bone struct?

A few reasons:

The first reason is that the helper functions may be called using multiple different transforms, like the newly introduced `local_pose_override` in my draft PR or the existing `global_pose_override`. Ultimately, it would probably only ever by the two override poses that ever get used with these functions, but then the Bone struct would need an additional two transforms.

The second reason is that including it in the Bone struct would increase the computations for those using just the Skeleton3D node. If possible, I figure it is best to try and make the IK code mostly separate and limit the performance impact it could have for those not using it.

The third reason is that, with how the IK needs to be run, it would be recalculated more often than needed as the bones in the chain are iteratively updated. Keeping it a separate function means it is only used when needed. Especially for iterative IK, like FABRIK, CCDIK, etc, the cost could add up quickly when calculating it every time one of the bones changes. Keeping it out of the Bone struct avoids this issue entirely.

Reply:
I'm not sure which is faster, keeping the data in the bones struct vs going from the bone all the way to the parent bone every time the function is called. 

Probably need to test it. 

We thought that recreating all the transforms from the bone all the way to the bone chain root would be expensive, but it's really better done as a bench mark with 300-500 bones.

You do need to iterate to the root of the bone chain operation, but as mentioned it's undecided keeping the data as part of the struct is slower than recomputing every function call.